psi4/src/psi4/libmints/matrix.h

```diff
@@ -593,6 +593,9 @@ class Matrix : public std::enable_shared_from_this<Matrix> {
     const Dimension& rowspi() const {
         return rowspi_;
     }
+    Dimension& rowspi() {
+        return rowspi_;
+    }
```

Comment:
As above, I wouldn't allow non-const Dimension objects. A user modifying one of these objects in place could create seriously hard to track down bugs.